contracts/contracts/solidity/StarknetValidator.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >0.6.0 <0.9.0;
+
+import '@chainlink/contracts/src/v0.8/interfaces/AggregatorValidatorInterface.sol';
+import '@chainlink/contracts/src/v0.8/interfaces/TypeAndVersionInterface.sol';
+import '@chainlink/contracts/src/v0.8/interfaces/AccessControllerInterface.sol';
+import '@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol';
+import '@chainlink/contracts/src/v0.8/SimpleWriteAccessController.sol';
+
+import '@chainlink/contracts/src/v0.8/dev/vendor/openzeppelin-solidity/v4.3.1/contracts/utils/Address.sol';
+
+import '../../vendor/starkware-libs/starkgate-contracts-solidity-v0.8/contracts/starkware/starknet/solidity/IStarknetMessaging.sol';
+
+/**
+ * @title Validator - makes cross chain call to update the Sequencer Uptime Feed on L2
+ */
+contract Validator is TypeAndVersionInterface, AggregatorValidatorInterface, SimpleWriteAccessController {
+    int256 private constant ANSWER_SEQ_OFFLINE = 1;
+    /* Selector hardcoded because StarkNet generates selectors differently than the standard ethereum way
+    different hash function on stark curve is used */
+    uint256 constant STARK_SELECTOR_UPDATE_STATUS =
+        1585322027166395525705364165097050997465692350398750944680096081848180365267;
+
+    IStarknetMessaging public immutable STARKNET_MESSAGING;
+    uint256 public L2_UPTIME_FEED_ADDR;
+
+    /**
+     * @param starknetMessaging the address of the Starknet Messaging contract address
+     * @param l2UptimeFeedAddr the address of the Sequencer Uptime Feed on L2
+     */
+    constructor(address starknetMessaging, uint256 l2UptimeFeedAddr) {
+        require(starknetMessaging != address(0), 'Invalid Starknet Messaging address');
+        require(l2UptimeFeedAddr != 0, 'Invalid StarknetSequencerUptimeFeed contract address');
+        STARKNET_MESSAGING = IStarknetMessaging(starknetMessaging);
+        L2_UPTIME_FEED_ADDR = l2UptimeFeedAddr;
+    }
+
+    /**
+     * @notice toUInt256 convert a bool to uint256.
+     */
+    function toUInt256(bool x) internal pure returns (uint256 r) {
```

Comment:
@krebernisak is there a better way to do this conversion?

Reply:
Not sure there is: https://github.com/ethereum/solidity/issues/12930